Fix compatibility with binaries that reference __stack_chk_guard
TODO: find where these references are coming from.

Index: glibc-2.32/sysdeps/mach/hurd/dl-sysdep.c
===================================================================
--- glibc-2.32.orig/sysdeps/mach/hurd/dl-sysdep.c
+++ glibc-2.32/sysdeps/mach/hurd/dl-sysdep.c
@@ -68,6 +68,8 @@ void *_dl_random attribute_relro = NULL;
 
 struct hurd_startup_data *_dl_hurd_data;
 
+uintptr_t __stack_chk_guard attribute_relro;
+
 
 ElfW(Addr)
 _dl_sysdep_start (void **start_argptr,
Index: glibc-2.32/sysdeps/mach/hurd/i386/ld.abilist
===================================================================
--- glibc-2.32.orig/sysdeps/mach/hurd/i386/ld.abilist
+++ glibc-2.32/sysdeps/mach/hurd/i386/ld.abilist
@@ -17,3 +17,4 @@ GLIBC_2.2.6 _r_debug D 0x14
 GLIBC_2.3 ___tls_get_addr F
 GLIBC_2.3 __tls_get_addr F
 GLIBC_2.34 __rtld_version_placeholder F
+GLIBC_2.4 __stack_chk_guard D 0x4
Index: glibc-2.38/sysdeps/mach/hurd/x86_64/ld.abilist
===================================================================
--- glibc-2.38/sysdeps/mach/hurd/x86_64/ld.abilist.original	2024-02-19 00:30:47.000000000 +0000
+++ glibc-2.38/sysdeps/mach/hurd/x86_64/ld.abilist	2024-02-19 08:48:42.000000000 +0000
@@ -8,6 +8,7 @@
 GLIBC_2.38 __pread64 F
 GLIBC_2.38 __read F
 GLIBC_2.38 __sbrk F
+GLIBC_2.38 __stack_chk_guard D 0x8
 GLIBC_2.38 __tls_get_addr F
 GLIBC_2.38 __write F
 GLIBC_2.38 __writev F
